DEVELOPMENT.md

```diff
@@ -0,0 +1,88 @@
+## Run services with `go run`
+
+This section explains how to run some of the services using `go run` instead of the Docker setup to facilitate development.
+
+### Running `sda-download` with `go run`
+1. Bring up all SDA services with the S3 backend and populate them with test data by running the following command in the root folder of the repository:
+```sh
+make integrationtest-sda-s3-run 
+```
+
+2. Change to the folder `sda-download` and start the `sda-download` service using:
+```sh
+CONFIGFILE=dev_utils/config-notls_local.yaml go run cmd/main.go
+```
+> If the host `oidc` cannot be accessed on the local machine, update your `/etc/hosts` file by adding the following line to ensure the service can resolve `oidc` locally:
+> ```
+> 127.0.0.1 oidc
+> ```
+
+3. Check if `sda-download` works as expected using:
+```sh
+curl -o /dev/null -s -w "%{http_code}\n" http://localhost:18080/health
+```
+If successful, the curl command should output the HTTP code `200`.
+
+You can further check the endpoint `/metadata/datasets` using:
+```sh
+token=$(curl -s -k http://localhost:8080/tokens | jq -r '.[0]') 
+curl -H "Authorization: Bearer $token" http://localhost:18080/metadata/datasets
+```
+If successful, the curl command should output a JSON body containing:
+```json
+["EGAD74900000101"]
+```
+
+
+### Running other SDA services with `go run`
+Running other SDA services located in the `sda` subfolder, such as `ingest` or `verify`, differs slightly from running the `sda-download` service. Here, we'll use `ingest` as an example.
+
+1. Bring up all SDA services with the S3 backend and populate them with test data by running the following command in the root folder of the repository:
+```sh
+make integrationtest-sda-s3-run 
```

Comment:
```suggestion
make sda-s3-up
```

This should probably use just the startup command that is part of PR #1162  since this will start the test script and shutting down ingest later will cause it to fail.

Reply:
If you just bring up the services with `sda-s3-up`, the verification test mentioned in the last point will not work. Also, if you wait until `make integrationtest-sda-s3-run` is finished, running docker stop ingest will not cause it to fail. Perhaps I should add a note to wait until the command finishes?

Reply:
I create a bug report https://github.com/neicnordic/sensitive-data-archive/issues/1184

Reply:
It's not a bug, it's you who haven't registered the key hash in the DB before you try to ingest the file.

Reply:
Thanks for pointing this out. Fixed in the commit 2cefebe